### PR TITLE
An extremely large pull request with lots of changes

### PR DESCRIPTION
**Pull request by adon-cloudcannon**
# Context

WIP - apologies for the rather large PR - this began as a bit of a play as part of my onboarding, but I think is significant enough to be merged in now... will try to feature branch and separate code better in the future!

[Link to Notion ticket](https://www.notion.so/cloudcannon/11ty-Starter-Template-7abb3ae1482c4960a0eabc50cd808503?pvs=4)

# Additional information

This was originally a template based on a client site. A main focus has been stripping out client-specific content/images etc...

This code features:

- A redeveloped left-right component that allows the user to switch the position of the image.
- A generic heading-text component that is used in both left-right component and hero.
- A newly developed infinite image carousel component (WIP but functional).
- A custom postcss plugin to process all css or scss files throughout the project (bookshop styles included) and output legal CSS
- Also, a custom eleventy plugin that automatically runs the postcss - thus, we do away with custom commands in the package.json... everything just runs on build:eleventy or watch:eleventy

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component(s) in Bookshop browser
- The site in CloudCannon [Link here](https://deep-car.cloudvent.net/)

### CODE REVIEW

**Comment by adon-cloudcannon** [comment]
@tomrcc - I have a few PRs on different branches... um, they might be a little all over the place since this PR has lots of random changes... and since then I've tried to branch features, but a few of them are probably a little out of sync with main, etc...

If we start merging them in, it should (hopefully) all sync up again... but just a heads up that I think this first one might mess something up a little later... :/

File: .DS_Store
Comment (tomrcc) [inline .DS_Store:1]: Add this to .gitignore

File: .eleventy.js
Comment (tomrcc) [inline .eleventy.js:1]: Can we add comments to assign_local explaining what it's doing?

File: component-library/text.scss
Comment (tomrcc) [inline component-library/text.scss:1]: What this do?

File: utils/DEFUNCT.cloudcannon-postcss.js
Comment (tomrcc) [inline utils/DEFUNCT.cloudcannon-postcss.js:1]: What this? What's the significance of DEFUNCT?

**Review by tomrcc — APPROVED** [review]
Couple small changes, and a couple of questions.
Happy for you to merge this in. 
Some simple documentation in the template README.md about what's happening with PostCSS might be a nice thing to have eventually.

File: component-library/text.scss
Comment (adon-cloudcannon) [reply component-library/text.scss:1]: Nothing :)
Removing...

File: utils/DEFUNCT.cloudcannon-postcss.js
Comment (adon-cloudcannon) [reply utils/DEFUNCT.cloudcannon-postcss.js:1]: Nothing :)
Removing...

File: .DS_Store
Comment (adon-cloudcannon) [reply .DS_Store:1]: Done

File: .eleventy.js
Comment (adon-cloudcannon) [reply .eleventy.js:1]: Will check with Alysha to see if there're some existing comments we can steal... just to make sure I explain exactly what it's doing correctly - future PR

**Pull request merged**